src/Run.js

```diff
@@ -37,8 +42,15 @@ function Run(ref, json_q, callback,ind, progressObj) {
       // Tell UI that a step is being drawn.
       ref.images[image].steps[i].UI.onDraw(ref.images[image].steps[i].options.step);
 
+      var inputForNextStep = ref.copy(input);
```

Comment:
Let's break this out into another file called something like `RunToolkit` or something... so we have a very distinct moment where we control and curate the tools we make available?